### PR TITLE
mv test corr

### PR DESCRIPTION
**Pull request by jvdp1**
@awvwgk : I move test_corr.f90 to test_corr.fypp for using the new testing approach

### CODE REVIEW

File: src/tests/stats/test_corr.fypp [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+contains
+
+    !> Collect all exported unit tests
+    subroutine collect_stats_corr(testsuite)
+        !> Collection of tests
+        type(unittest_type), allocatable, intent(out) :: testsuite(:)
+
+        testsuite = [ &
+            new_unittest("test_${k1}$_1dim", test_${k1}$_1dim) &
```
Comment (awvwgk) [inline src/tests/stats/test_corr.fypp:50]: Is this intended? `k1` would be undefined here, wouldn't it?

Comment (jvdp1) [reply src/tests/stats/test_corr.fypp:50]: No. you are right. It should be replaced by `sp`.

**Comment by awvwgk** [comment]
Any idea why this fails on 32bit Windows?

**Comment by jvdp1** [comment]
> Any idea why this fails on 32bit Windows?

No. I checked the logs, and it is a Segfault (invalid memory reference).

**Pull request closed without merging**